src/main/java/org/team1540/robot2025/subsystems/arm/ArmIOReal.java

```diff
@@ -0,0 +1,134 @@
+
+    private final MotionMagicVoltage positionCtrlReq = new MotionMagicVoltage(0).withSlot(0);
+    private final VoltageOut voltageCtrlReq = new VoltageOut(0);
+
+    // constructor
+    public ArmIOReal() {
+        TalonFXConfiguration motorConfig = new TalonFXConfiguration();
+
+        motorConfig.MotorOutput.NeutralMode = NeutralModeValue.Coast;
+        motorConfig.MotorOutput.Inverted = InvertedValue.Clockwise_Positive;
+
+        motorConfig.Feedback.FeedbackSensorSource = FeedbackSensorSourceValue.FusedCANcoder;
+        motorConfig.Feedback.FeedbackRemoteSensorID = CANCODER_ID;
+        motorConfig.Feedback.SensorToMechanismRatio = CANCODER_TO_PIVOT_RATIO * MOTOR_TO_CANCODER;
+        motorConfig.Feedback.RotorToSensorRatio = 1;
+
+        motorConfig.SoftwareLimitSwitch.ForwardSoftLimitEnable = true;
+        motorConfig.SoftwareLimitSwitch.ForwardSoftLimitThreshold = MAX_ANGLE.getRotations();
+        motorConfig.SoftwareLimitSwitch.ReverseSoftLimitEnable = true;
+        motorConfig.SoftwareLimitSwitch.ForwardSoftLimitThreshold = MIN_ANGLE.getRotations();
+
+        motorConfig.Slot0.kP = KP;
+        motorConfig.Slot0.kI = KI;
+        motorConfig.Slot0.kD = KD;
+        motorConfig.Slot0.kS = KS;
+        motorConfig.Slot0.kG = KG;
+        motorConfig.Slot0.kV = KV;
+        motorConfig.Slot0.GravityType = GravityTypeValue.Arm_Cosine;
+        /*
+        Note that the sensor offset and ratios must be
+        configured so that the sensor reports a position
+         of 0 when the mechanism is horizonal
+         (parallel to the ground), and the reported
+         sensor position is 1:1 with the mechanism.
+         */
+
+        motorConfig.MotionMagic.MotionMagicCruiseVelocity = CRUISE_VELOCITY_RPS;
+        motorConfig.MotionMagic.MotionMagicAcceleration = MAX_ACCEL_RPS;
+        motorConfig.MotionMagic.MotionMagicJerk = JERK_RPS;
+
+        motorConfig.CurrentLimits.SupplyCurrentLimitEnable = true;
+        motorConfig.CurrentLimits.SupplyCurrentLimit = CURRENT_LIMIT;
+        motorConfig.CurrentLimits.SupplyCurrentLowerLimit = CURRENT_LOWER_LIMIT;
+        motorConfig.CurrentLimits.SupplyCurrentLowerTime = TIME_LOWER_LIMIT;
+
+        CANcoderConfiguration cancoderConfig = new CANcoderConfiguration();
+        cancoderConfig.MagnetSensor.SensorDirection = SensorDirectionValue.CounterClockwise_Positive;
+        cancoderConfig.MagnetSensor.MagnetOffset = CANCODER_OFFSET_ROTS;
+        cancoderConfig.MagnetSensor.AbsoluteSensorDiscontinuityPoint = DISCONTINUITY_POINT;
+
+        cancoder.getConfigurator().apply(cancoderConfig);
+        motor.getConfigurator().apply(motorConfig);
+        motor.setPosition(position.getValueAsDouble() * MOTOR_TO_CANCODER);
+        BaseStatusSignal.setUpdateFrequencyForAll(
+                50, position, velocity, appliedVoltage, current, temp, forwardLimit, reverseLimit);
+
+        motor.optimizeBusUtilization();
+        cancoder.optimizeBusUtilization();
+        motor.setPosition(
```

Comment:
You shouldn't need to update the motor position manually, it'll talk to the cancoder to figure stuff out

https://v6.docs.ctr-electronics.com/en/latest/docs/api-reference/device-specific/talonfx/remote-sensors.html